### ISSUE TITLE
Headless update command not working

### ISSUE THREAD

**Issue by xfzv** [RubixDev/mason-update-all#2]
```console
❯ nvim --version
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3
```

I'm using `mason-update-all` with Lazy:

```lua
{
    "RubixDev/mason-update-all",
    config = true,
},
```

When trying to update Mason packages (I have 18 of them installed) with the following:

https://github.com/RubixDev/mason-update-all/blob/d94cf0ff32a5469ca887dd99914bf4b095ebc24f/README.md?plain=1#L61-L63

I'm getting different outputs and most of the time an error:

```console
❯ nvim --headless -c 'autocmd User MasonUpdateAllComplete quitall' -c 'MasonUpdateAll'
[mason-update-all] Fetching updates
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to updateError detected while processing User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete":
E218: autocommand nesting too deep
E218: autocommand nesting too deep%


❯ nvim --headless -c 'autocmd User MasonUpdateAllComplete quitall' -c 'MasonUpdateAll'
[mason-update-all] Fetching updates
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update%


❯ nvim --headless -c 'autocmd User MasonUpdateAllComplete quitall' -c 'MasonUpdateAll'
[mason-update-all] Fetching updates
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to update
[mason-update-all] Nothing to updateError detected while processing User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete"..User Autocommands for "MasonUpdateAllComplete":
E218: autocommand nesting too deep%
```

After that, when running `:Mason` directly from Neovim, some packages could still be updated. No issue when running directly `:MasonUpdateAll` from Neovim.

Any idea?

---

Edit: I have no issue with other headless commands such as:

```console
# Lazy plugins
nvim --headless "+Lazy! sync" +qa

# Treesitter parsers
nvim --headless -c "TSUpdateSync" +qa
```

**Comment by run2cmd** [comment]
I've created https://github.com/RubixDev/mason-update-all/pull/3 to address this.

**Comment by RubixDev** [comment]
This should be fixed in 68d9d893b039a870c3400cde17fb0485f9934c3e
